backend/tex/aux_parser.py

```diff
@@ -0,0 +1,36 @@
+import argparse
+import json
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--file', help='filepath to input aux')
+args = parser.parse_args()
+
+f = open(args.file, 'r')
+s = '\\newlabel'
+
+listOfLabels = []
+
+def getArg(s, i):
+    end = s[i+1:].find("}")
+    return s[i+1:end + i + 1], end + i + 2
+
+for line in f:
+    if line.startswith(s):
+        line = line.replace(s, '')
+        caption, ind = getArg(line, 0)
```

Comment:
do you really need to make the first arg so exceptional? Why don't you parse the whole line of nested labels and use its 0 item as caption?